### PR TITLE
Do not call get_user_id if it is not needed

### LINKED ISSUE #356: Windows : ImportError: No module named pwd on util.py
When i try to build gitpython on Windows Server i have this error message.

```powershell
File "C:\project\.tox\back\lib\site-packages\git\objects\commit.py", line 318, in create_from_tree
    committer = committer or Actor.committer(cr)
  File "C:\project\.tox\back\lib\site-packages\git\util.py", line 416, in committer
    return cls._main_actor(cls.env_committer_name, cls.env_committer_email, config_reader)
  File "C:\project\.tox\back\lib\site-packages\git\util.py", line 386, in _main_actor
    default_email = get_user_id()
  File "C:\project\.tox\back\lib\site-packages\git\util.py", line 152, in get_user_id
    return "%s@%s" % (getpass.getuser(), platform.node())
  File "c:\python27\Lib\getpass.py", line 157, in getuser
    import pwd
ImportError: No module named pwd
```

After search, the only solution is try [this hack](https://t37.net/on-getting-the-running-process-user-under-python-and-supervisord.html) on [utiy.py](https://github.com/gitpython-developers/GitPython/blob/master/git/util.py#L14) file.

Then use this :  

```python
try:
    import pwd
except ImportError:
    import winpwd as pwd
import os

def get_user_id():
    """:return: string identifying the currently active system user as name@node"""
    return "%s@%s" % (pwd.getpwuid(os.getuid()).pw_name, platform.node())
```

instead of 

```python
import getpass

def get_user_id():
    """:return: string identifying the currently active system user as name@node"""
    return "%s@%s" % (getpass.getuser(), platform.node())
```

What do you think about that ?

### PR DESCRIPTION
**Pull request by eric-wieser**
On systems without any environment variables (such as sandboxed CI runs) and no `pwd` module (such as windows), GitPython crashes.

This is just a logic error - there is no reason to fallback to `pwd` until the config lookup has actually failed. The implementation of `ConfigWriter.get_value` is to catch `Exception`, so this patch doesn't attempt to work out a more specific exception than what was used before.

This fixes #356

### CODE REVIEW

**Comment by eric-wieser** [comment]
It looks like a maintainer might have to approve the workflow runs each time I try to fix the mypy troubles...

**Comment by Yobmod** [comment]
I've added an overload to config.py  get_value() in main that fixes a mypy issue

**Comment by eric-wieser** [comment]
> And I merged the other PR, so hopefully the checks on this one will auto run

No you didn't, you closed it!

File: test/test_util.py
Comment (eric-wieser) [inline test/test_util.py:243]: I assume there's some way to get a writeable repo that we can override the config in, but I'm not familiar enough with your fixtures to want to go to the effort of working that out.

Comment (Byron) [reply test/test_util.py:243]: Thanks a lot for helping with the test. Getting a repository with write permissions [can be done like this](https://github.com/gitpython-developers/GitPython/blob/main/test/test_base.py#L104:L107).

Comment (eric-wieser) [reply test/test_util.py:261]: Which test fails?

Comment (Yobmod) [reply test/test_util.py:261]: Test_remote.py, assertion at line 333 failed.

Also came with warnings about not being able to create temporary folder at git://localhost:19418/daemon_repo-test_base-5tjbtmwj.

But fixed by setting git.email --global

Comment (eric-wieser) [reply test/test_util.py:261]: Oh, I thought you meant my new test failed!

**Review by Byron — APPROVED** [review]
Thank a lot, this looks good to me.

It seems the difficulty stems more from dealing with tests and CI than the actual code change, and I don't fully follow. Hence I leave it to @Yobmod to merge in case there is nothing I missed.

**Pull request merged**
